x-pack/plugins/graph/public/components/field_manager/field_editor.tsx

```diff
@@ -71,6 +71,14 @@ export function FieldEditor({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [initialField]);
 
+  // In case of cleared field and the user closes the popover, restore the initial field
```

Comment:
I don't think we need this. Currently the popover stores the current edit state as long as the user doesn't hit cancel (for example change the color, then click outside of the popover, then open it again - the selection will be retained).

I don't see why we would need to do something else for the case of empty field

Reply:
I went with that at first. But then trying to test it, I found a bit annoying that the field was still empty and not reset.
I understand that it's completely subjective and do not really have a strong opinion on that. :)

Reply:
An alternative would be to reset the whole form (like when hitting "Cancel") on popover closing. This will do the same thing (in a consistent way), but won't introduce a special code path for this specific case.

Reply:
yeah, it makes sense. I'll change it this way 👍